support-frontend/assets/components/subscriptionCheckouts/addressSearch/addressComponentStore.js

```diff
@@ -0,0 +1,262 @@
+
+  return [...homeRules, ...billingRules];
+};
+
+// ----- Action Creators ----- //
+
+const setFormErrorsFor = (scope: AddressType) => (errors: Array<FormError<FormField>>): Action => ({
+  scope,
+  type: 'SET_ADDRESS_FORM_ERRORS',
+  errors,
+});
+
+const addressActionCreatorsFor = (scope: AddressType) => ({
+  setFormErrors: (errors: FormError<FormField>[]) =>
+    (dispatch: Dispatch<Action>) => dispatch(setFormErrorsFor(scope)(errors)),
+  setCountry: (countryRaw: string) => (dispatch: Dispatch<SetCountryChangedAction | SetCountryAction>) => {
+    const country = fromString(countryRaw);
+    if (country) {
+      dispatch(setCountry(country));
+      dispatch({
+        type: 'SET_COUNTRY_CHANGED',
+        country,
+        scope,
+      });
+    }
+  },
+  setAddressLineOne: (lineOne: string): Function => (setFormSubmissionDependentValue(() => ({
+    scope,
+    type: 'SET_ADDRESS_LINE_1',
+    lineOne,
+  }))),
+  setAddressLineTwo: (lineTwo: string): Action => ({
+    scope,
+    type: 'SET_ADDRESS_LINE_2',
+    lineTwo,
+  }),
+  setTownCity: (city: string): Function => (setFormSubmissionDependentValue(() => ({
+    scope,
+    type: 'SET_TOWN_CITY',
+    city,
+  }))),
+  setState: (state: string): Function => (setFormSubmissionDependentValue(() => ({
+    type: 'SET_STATE',
+    state,
+    scope,
+  }))),
+  setPostcode: (postCode: string): Function => (setFormSubmissionDependentValue(() => ({
+    type: 'SET_POSTCODE',
+    postCode,
+    scope,
+  }))),
+});
+
+export type ActionCreators = $Call<typeof addressActionCreatorsFor, AddressType>;
+
+// ----- Reducer ----- //
+
+function addressReducerFor(scope: AddressType, initialCountry: IsoCountry) {
+
+  const initialState = {
+    country: initialCountry,
+    lineOne: null,
+    lineTwo: null,
+    city: null,
+    postCode: null,
+    state: null,
+    formErrors: [],
+  };
+
+  const fields = (state: AddressFieldsState = initialState, action: Action): AddressFieldsState => {
+
+    if (action.scope !== scope) {
+      return state;
+    }
+
+    switch (action.type) {
+
+      case 'SET_ADDRESS_LINE_1':
+        return { ...state, formErrors: removeError('lineOne', state.formErrors), lineOne: action.lineOne };
+
+      case 'SET_ADDRESS_LINE_2':
+        return { ...state, lineTwo: action.lineTwo };
+
+      case 'SET_TOWN_CITY':
+        return { ...state, formErrors: removeError('city', state.formErrors), city: action.city };
+
+      case 'SET_STATE':
+        return { ...state, formErrors: removeError('state', state.formErrors), state: action.state };
+
+      case 'SET_POSTCODE':
+        return { ...state, formErrors: removeError('postCode', state.formErrors), postCode: action.postCode };
+
+      case 'SET_ADDRESS_FORM_ERRORS':
+        return { ...state, formErrors: action.errors };
+
+      case 'SET_COUNTRY_CHANGED':
+        return {
+          ...state,
+          state: null,
+          formErrors: [],
```

Comment:
if you change the country we remove all the error but we only clear the state field rather than all the address fields, is that right here?

Reply:
Yes that's how it works and I think it makes sense